Field to toggle auth request signing

kadaji opened the pull request:
By default the auth request signing field is set to False. Added a checkbox to allow configuring the setting.

kadaji committed:
setting authnRequestsSigned to True

kadaji committed:
added field to toggle authentication request signing

dcwatson commented on sp/admin.py:
```diff
@@ -51,6 +51,7 @@ class IdPAdmin(admin.ModelAdmin):
                 "fields": (
                     "contact_name",
                     "contact_email",
+                    "authn_requests_signed",
```
This should be down in the "Advanced" group below.

dcwatson commented on sp/models.py:
```diff
@@ -44,6 +44,11 @@ class IdP(models.Model):
     contact_name = models.CharField(max_length=100)
     contact_email = models.EmailField(max_length=100)
+
+    authn_requests_signed = models.BooleanField(
```
Minor quibble, but can you put this down by the `logout_request_signed` and `logout_response_signed` fields?

dcwatson commented on sp/models.py:
```diff
@@ -217,6 +222,7 @@ def sp_settings(self):
             "security": {
                 "wantAttributeStatement": self.require_attributes,
                 "metadataValidUntil": self.certificate_expires,
+                "authnRequestsSigned": self.,
```
I think you missed something here 😄

dcwatson commented on sp/views.py:
```diff
@@ -13,6 +13,8 @@
 
 def metadata(request, **kwargs):
     idp = get_request_idp(request, **kwargs)
+    print(idp.sp_settings)
```
Let's remove this

kadaji committed:
fixed missing auth request value

akadajis-su committed:
feat(sp): add WantAssertionsSigned; reflect both signing flags in metadata

akadajis-su committed:
feat(sp): log all SAML response attributes (one row per response)

akadajis-su committed:
feat(sp): add MyCESAMLAuthenticationBackend with per-IdP admin-error-email toggle

akadajis-su committed:
feat(sp): per-IdP custom authentication-failed message

akadajis-su committed:
feat(sp): admin action to duplicate an IdP configuration

akadajis-su committed:
fix(sp): review findings — best-effort attribute logging, backend nameid guard, backend tests, atomic duplicate

akadajis-su committed:
feat(sp): per-IdP toggle to disable response-attribute logging

Adds IdP.log_response_attributes (default True); the ACS view only writes
an IdPAttributeLog row when it is enabled, so IdPs whose attribute payload
should not be retained can opt out. Addresses PII-retention review finding.

akadajis-su committed:
fix(sp): review round 2 — audit nameid, skip test-flow logging, backend error handling, verify-branch message, read-only log, safe duplicate

akadajis-su committed:
fix(sp): review round 3 — deep-copy JSON fields in duplicate(), de-vacuous toggle test, document backend error propagation

kadaji committed:
Merge pull request #1 from Canusia/feature/myce-customizations

MyCE customizations: attribute logging, duplicate action, signing flags, auth toggles

akadajis-su committed:
chore(sp): bump version to 0.9.0 (MyCE customizations release)

kadaji committed:
feat(sp): export selected IdP configuration(s) to JSON (admin action)

kadaji committed:
feat(sp): import IdP configuration from pasted JSON (admin view, upsert by entity_id)

kadaji committed:
test(sp): round-trip export/import config restoration

Verifies export_idp_json -> _import_one restores name, signing flags,
private_key, attributes, user_defaults, and certificate_expires
(DateTimeField str()-on-export -> to_python()-on-import round trip).

kadaji committed:
fix(sp): guard non-object entries in JSON import to avoid AttributeError/500

A top-level payload that parses as valid JSON but isn't an object (or
array containing a non-object element) reached _import_one()/entry.get()
in the per-entry error handler, raising AttributeError and surfacing as
an HTTP 500 instead of a counted failure.

kadaji committed:
chore(sp): release v0.10.0 (IdP config JSON export/import)
